app/assets/javascripts/map.js

```diff
@@ -1,152 +1,165 @@
 //= require polyline
 
-      var directionsDisplay;
-      var directionsService = new google.maps.DirectionsService();
-      var map;
+var directionsService = new google.maps.DirectionsService();
 
-      function initialize() {
-        directionsDisplay = new google.maps.DirectionsRenderer();
+function initialize() {
+  directionsDisplay = new google.maps.DirectionsRenderer();
 
-        var mapOptions = {
-          zoom: 4,
-          mapTypeId: google.maps.MapTypeId.ROADMAP
-        };
+  var mapOptions = {
+    zoom: 4,
+    mapTypeId: google.maps.MapTypeId.ROADMAP
+  };
 
-        map = new google.maps.Map(document.getElementById('map_canvas'), mapOptions);
-        directionsDisplay.setMap(map);
+  map = new google.maps.Map(document.getElementById('map_canvas'), mapOptions);
+  directionsDisplay.setMap(map);
+  route = new Route();
+  route.calculateRoute();
 
-        calcRoute();
-      }
+}
 
-      function calcRoute() {
+//-------Route----------
 
-        var start = new google.maps.LatLng(41.953819, -87.654750); // Chicago
-        var end = new google.maps.LatLng(38.637548, -90.205010); // St. Louis
-        var waypts = [];
-        var checkboxArray = document.getElementById('waypoints');
+var Route = function(){
+  this.start = new google.maps.LatLng(41.953819, -87.654750), // Chicago for now
```

Comment:
This is called a "magic number."  You might consider a constant like

`CHICAGO_LONGITUDE` or `CHICAGO_LATITUDE`.  This allows you to store all these "temp" declarations in one place (and makes it easier to adjust).